### PR TITLE
Add timeout to agent watches

### LINKED ISSUE #4134: Agent address watches don't honour resync timeout
**Describe the bug**

Testing release-0.31 ion CRC, I notice that the address watches initiated by agent last only 90-120seconds even though the intent is for a 600 second timeout to apply.   This problem doesn't occur with the agent's pod watch.

This issue may mean that the admin pod consumes more resource (CPU) and imposes more load on the Kubernetes API than is necessary.  This may be particularly apparent with many addresses.

This can be seen from the logs:

```
2020-03-17T20:25:23.443Z agent INFO  GET /apis/enmasse.io/v1beta1/watch/namespaces/enmasse-infra/addresses => 200
2020-03-17T20:26:38.768Z agent INFO  GET /apis/enmasse.io/v1beta1/namespaces/enmasse-infra/addresses => 200
2020-03-17T20:26:38.771Z agent INFO  GET /apis/enmasse.io/v1beta1/watch/namespaces/enmasse-infra/addresses => 200
2020-03-17T20:27:50.044Z agent INFO  GET /apis/enmasse.io/v1beta1/namespaces/enmasse-infra/addresses => 200
2020-03-17T20:27:50.048Z agent INFO  GET /apis/enmasse.io/v1beta1/watch/namespaces/enmasse-infra/addresses => 200
2020-03-17T20:29:23.162Z agent INFO  GET /apis/enmasse.io/v1beta1/namespaces/enmasse-infra/addresses => 200
2020-03-17T20:29:23.173Z agent INFO  GET /apis/enmasse.io/v1beta1/watch/namespaces/enmasse-infra/addresses => 200
2020-03-17T20:30:59.213Z agent INFO  GET /apis/enmasse.io/v1beta1/namespaces/enmasse-infra/addresses => 200
2020-03-17T20:30:59.215Z agent INFO  GET /apis/enmasse.io/v1beta1/watch/namespaces/enmasse-infra/addresses => 200
2020-03-17T20:32:14.028Z agent INFO  GET /apis/enmasse.io/v1beta1/namespaces/enmasse-infra/addresses => 200
2020-03-17T20:32:14.033Z agent INFO  GET /apis/enmasse.io/v1beta1/watch/namespaces/enmasse-infra/addresses => 200
```

compared with:

```
2020-03-17T19:45:59.426Z agent INFO  GET /api/v1/watch/namespaces/enmasse-infra/pods?labelSelector=role%3Dbroker%2CinfraUuid%3Dqueuespace => 200
2020-03-17T19:55:59.427Z agent INFO  response /api/v1/watch/namespaces/enmasse-infra/pods?labelSelector=role%3Dbroker%2CinfraUuid%3Dqueuespace timeout (600s)
2020-03-17T19:55:59.433Z agent INFO  GET /api/v1/namespaces/enmasse-infra/pods?labelSelector=role%3Dbroker%2CinfraUuid%3Dqueuespace => 200
2020-03-17T19:55:59.436Z agent INFO  GET /api/v1/watch/namespaces/enmasse-infra/pods?labelSelector=role%3Dbroker%2CinfraUuid%3Dqueuespace => 200
2020-03-17T20:05:59.439Z agent INFO  response /api/v1/watch/namespaces/enmasse-infra/pods?labelSelector=role%3Dbroker%2CinfraUuid%3Dqueuespace timeout (600s)
2020-03-17T20:05:59.451Z agent INFO  GET /api/v1/namespaces/enmasse-infra/pods?labelSelector=role%3Dbroker%2CinfraUuid%3Dqueuespace => 200
2020-03-17T20:05:59.458Z agent INFO  GET /api/v1/watch/namespaces/enmasse-infra/pods?labelSelector=role%3Dbroker%2CinfraUuid%3Dqueuespace => 200
2020-03-17T20:15:59.460Z agent INFO  response /api/v1/watch/namespaces/enmasse-infra/pods?labelSelector=role%3Dbroker%2CinfraUuid%3Dqueuespace timeout (600s)
2020-03-17T20:15:59.468Z agent INFO  GET /api/v1/namespaces/enmasse-infra/pods?labelSelector=role%3Dbroker%2CinfraUuid%3Dqueuespace => 200
2020-03-17T20:15:59.474Z agent INFO  GET /api/v1/watch/namespaces/enmasse-infra/pods?labelSelector=role%3Dbroker%2CinfraUuid%3Dqueuespace => 200
2020-03-17T20:25:59.476Z agent INFO  response /api/v1/watch/namespaces/enmasse-infra/pods?labelSelector=role%3Dbroker%2CinfraUuid%3Dqueuespace timeout (600s)
2020-03-17T20:25:59.486Z agent INFO  GET /api/v1/namespaces/enmasse-infra/pods?labelSelector=role%3Dbroker%2CinfraUuid%3Dqueuespace => 200
2020-03-17T20:25:59.491Z agent INFO  GET /api/v1/watch/namespaces/enmasse-infra/pods?labelSelector=role%3Dbroker%2CinfraUuid%3Dqueuespace => 200
```

**To Reproduce**

1. Create an standard address space and examine the agent log.

**Expected behavior**

Address watch should honour the 600s timeout.

**Additional context**

You can see the same behaviour from the command by comparing:

`oc get pods -w`
and 
`oc get address -w`

It seems that all CRDs have a lower timeout imposed by the server.

### PR DESCRIPTION
**Pull request by vbusch**
# Type of change

- Bugfix

### Description

Fixes #4134
Copies requestTimeout into timeoutSeconds on the query params

### Checklist

- [ ] Update/write design documentation in `./documentation/design`
- [ ] Write tests and make sure they pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

**Comment by vbusch** [comment]
@enmasse-ci run tests profile=isolated

File: agent/lib/kubernetes.js [hunk 1]
```diff
@@ -70,8 +70,15 @@ function get_options(options, path) {
...
     var path = base + namespace + '/' + resource;
+    var queryParams = {};
     if (options.selector) {
-        path += '?' + querystring.stringify({'labelSelector':options.selector});
+        queryParams.labelSelector = options.selector;
+    }
+    if (options.requestTimeout) {
+        queryParams.timeoutSeconds = options.requestTimeout;
+    }
+    if (Object.keys(queryParams).length) {
```
Comment (lulf) [inline agent/lib/kubernetes.js]: I know it works, but maybe use length > 0 just to help the reader :)

Comment (vbusch) [reply agent/lib/kubernetes.js]: Fix commited.

File: agent/lib/kubernetes.js [hunk 2]
```diff
@@ -85,7 +92,11 @@ function list_options(resource, baseOptions) {
 
 function watch_options(resource, baseOptions) {
     var base = resource.startsWith("address") ? '/apis/enmasse.io/v1beta1/watch/namespaces/' : '/api/v1/watch/namespaces/';
-    var options = get_options(baseOptions, get_path(base, resource, baseOptions));
+    var options = get_options(baseOptions, get_path(base, resource, includeResyncInterval(baseOptions)));
```
Comment (k-wall) [inline agent/lib/kubernetes.js]: I find applying resync interval twice distasteful and hard to follow.
WDYT about: https://github.com/vbusch/enmasse/pull/6/

Comment (vbusch) [reply agent/lib/kubernetes.js]: getOptions won't preserve the requestTimeout.
Does my latest commit help?  https://github.com/EnMasseProject/enmasse/pull/4137/commits/c056a1018cf6fafa62e83a9d02a13f2ef565d345

Comment (k-wall) [reply agent/lib/kubernetes.js]: Yes, it does.   thanks you.
